lib/api/salesforceChatAPI.js

```diff
@@ -0,0 +1,81 @@
+import { NativeModules } from "react-native";
+
+export default class SalesforceChatAPI {
+  constructor() {
+    this.salesforceChat = NativeModules.RNSalesforceChat;
+  }
+
+  throwRequiredFieldError(field) {
+    throw new Error(`required field ${field}`);
+  }
+
+  async createUserData({ agentLabel, value, isDisplayedToAgent }) {
+    if (!agentLabel) this.throwRequiredFieldError("agentLabel");
+    if (isDisplayedToAgent === undefined)
+      this.throwRequiredFieldError("isDisplayedToAgent");
+
+    this.salesforceChat.createPreChatData(
+      agentLabel,
+      value,
+      isDisplayedToAgent
+    );
+  }
+
+  async createEntityField({
+    objectFieldName,
+    doFind,
+    doCreate,
+    isExactMatch,
+    keyChatUserDataToMap,
+  }) {
+    if (!objectFieldName) this.throwRequiredFieldError("objectFieldName");
+    if (doFind === undefined) this.throwRequiredFieldError("doFind");
+    if (doCreate === undefined) this.throwRequiredFieldError("doCreate");
+    if (isExactMatch === undefined)
```

Comment:
@lenoirzamboni is it ok if these fields are `null`?

Reply:
@pedrohperalta Nope. I pushed some changes that also verify nullable boolean params. What do you think?